src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -682,6 +743,107 @@ function InfoBottomSheet({
   )
 }
 
+function ActionCard({ action }: { action: BeforeDepositAction }) {
+  return (
+    <Touchable
+      style={styles.touchable}
+      key={action.name}
+      borderRadius={20}
+      onPress={action.onPress}
+      testID={`Earn/BeforeDepositBottomSheet/${action.name}`}
+    >
+      <>
+        <action.iconComponent color={Colors.black} />
+        <View style={styles.cardContainer}>
+          <Text style={styles.actionTitle}>{action.title}</Text>
+          <Text style={styles.actionDetails}>{action.details}</Text>
+        </View>
+      </>
+    </Touchable>
+  )
+}
+
+function BeforeDepositBottomSheet({
+  forwardedRef,
+  token,
+  hasTokensOnSameNetwork,
+  hasTokensOnOtherNetworks,
+  canAdd,
+  exchanges,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  token: TokenBalance
+  hasTokensOnSameNetwork: boolean
+  hasTokensOnOtherNetworks: boolean
+  canAdd: boolean
+  exchanges: ExternalExchangeProvider[]
+}) {
+  const { t } = useTranslation()
+
+  const title = hasTokensOnSameNetwork
+    ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.youNeedTitle', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      })
+    : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.beforeYouCanDepositTitle')
+
+  const swapAndDepositAction = useSwapAndDepositAction({ token, forwardedRef })
+  const crossChainSwapAction = useCrossChainSwapAction({
+    token,
+    title: hasTokensOnSameNetwork
+      ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.action.crossChainSwap')
+      : t('addFundsActions.swap'),
+    details: hasTokensOnSameNetwork
+      ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.crossChainSwapActionDescription', {
+          tokenSymbol: token.symbol,
+        })
+      : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.swapActionDescription', {
+          tokenSymbol: token.symbol,
+          tokenNetwork: NETWORK_NAMES[token.networkId],
+        }),
```

Comment:
seems like this boolean is something that can be done inside of `useCrossChainSwapAction`